root/static/scripts/edit/components/AddButton.js

```diff
@@ -0,0 +1,38 @@
+/*
+ * @flow strict
+ * Copyright (C) 2024 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+component AddButton(
+  id: string,
+  onClick: (event: SyntheticEvent<HTMLButtonElement>) => void,
+  label?: string,
+) {
+  if (label == null) {
```

Comment:
Not sure I understand why this check is needed, since `{null}` as a child would output nothing anyway.

Reply:
For the `with-label` `className`, but I'm sure we can just move the check to be just inside `className` unless I'm forgetting something dumb, if it's fine to have `null` `{children}`?

Reply:
What about:

```
component AddButton(
  id: string,
  onClick: (event: SyntheticEvent<HTMLButtonElement>) => void,
  label?: string,
) {
  return (
    <button
      className={'add-item' + (nonEmpty(label) ? ' with-label' : '')}
      id={id}
      onClick={onClick}
      type="button"
    >
      {nonEmpty(label) ? label : null}
    </button>
  );
}
```